src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -10,143 +13,349 @@ module Dialogs
     # Part of {Sequences::AddPartition}.
     # Formerly MiniWorkflowStepPartitionSize
     class PartitionSize < CWM::Dialog
-      # @param disk []
-      # @param ptemplate []
-      # @param slots []
-      def initialize(disk, ptemplate, slots)
+      # @param disk_name [String]
+      # @param ptemplate [Sequences::PartitionTemplate]
+      #   a partition template, collecting data for a partition to be created
+      # @param regions [Array<Y2Storage::Region>]
+      #   regions available to create a partition in
+      def initialize(disk_name, ptemplate, regions)
         textdomain "storage"
-        @disk = disk
+        @disk_name = disk_name
         @ptemplate = ptemplate
-        @slots = slots
+        @regions = regions
       end
 
+      # @macro seeDialog
       def title
         # dialog title
-        Yast::Builtins.sformat(_("Add Partition on %1"), @disk.name)
+        Yast::Builtins.sformat(_("Add Partition on %1"), @disk_name)
       end
 
+      # @macro seeDialog
       def contents
-        VBox(SizeWidget.new(@disk, @ptemplate, @slots))
+        HVSquash(SizeWidget.new(@ptemplate, @regions))
       end
 
-      # Choose partition size: maximum, size based, start-end based.
-      class SizeWidget < CWM::CustomWidget
-        include Yast
-        include Yast::UIShortcuts
+      # Like CWM::RadioButtons but each RB has a subordinate indented widget.
+      # This is kind of like Pager, but all Pages being visible at once,
+      # and enabled/disabled.
+      # Besides {#items} you need to define also {#widgets}.
+      class ControllerRadioButtons < CWM::CustomWidget
+        def initialize
+          self.handle_all_events = true
+        end
+
+        # @return [WidgetTerm]
+        def contents
+          Frame(
+            label,
+            HBox(
+              HSpacing(hspacing),
+              RadioButtonGroup(Id(widget_id), buttons_with_widgets),
+              HSpacing(hspacing)
+            )
+          )
+        end
+
+        # @return [Numeric] margin at both sides of the options list
+        def hspacing
+          1.45
+        end
+
+        # @return [Numeric] margin above, between, and below the options
+        def vspacing
+          0.45
+        end
+
+        # @return [Array<Array(id,String)>]
+        abstract_method :items
+
+        # FIXME: allow {WidgetTerm}
+        # @return [Array<AbstractWidget>]
+        abstract_method :widgets
+
+        # @param event [Hash] UI event
+        def handle(event)
+          eid = event["ID"]
+          ids.zip(widgets).each do |id, widget|
+            if id == eid
+              widget.enable
+            else
+              widget.disable
```

Comment:
this won't work properly if any other widget event raised. It using handle_all_events, so it get all events from all widgets. it can be fixed with simple 
```
return nil unless ids.include?(eid)
ids.zip(widgets).each do |id, widget|
  ...
```